lib/core_ext/hash.rb

```diff
@@ -0,0 +1,8 @@
+class Hash
+  unless self.respond_to? :compact
+    # Simple method to remove nils from an array to ease use of the SOAP API
+    def compact
```

Comment:
Since this modifies the receiver, it should be called `compact!` to keep it consistent with other stdlib Hash/Array methods.